python/cvmfs/repository.py

```diff
@@ -198,38 +205,223 @@ def _pop_catalog(self):
 
 
 
-class Repository:
-    """ Abstract Wrapper around a CVMFS Repository representation """
+class Cache:
+    def __init__(self, cache_dir):
+        if not os.path.exists(cache_dir):
+            cache_dir = tempfile.mkdtemp(dir='/tmp', prefix='cache.')
+        self._cache_dir = cache_dir
+        self._create_cache_structure()
+        self._cleanup_metadata()
+
+    def _cleanup_metadata(self):
+        for f in glob.glob(self._cache_dir + '/' + '.cvmfs*'):
+            os.remove(f)
+
+
+    def _create_dir(self, path):
+        cache_full_path = os.path.join(self._cache_dir, path)
+        if not os.path.exists(cache_full_path):
+            os.mkdir(cache_full_path, 0755)
+
+
+    def _create_cache_structure(self):
+        self._create_dir('data')
+        for i in range(0x00, 0xff + 1):
+            new_folder = '{0:#0{1}x}'.format(i, 4)[2:]
+            self._create_dir(os.path.join('data', new_folder))
+
+
+    def cache_path(self):
+        return str(self._cache_dir)
+
+
+    def add(self, file_name):
+        full_path = os.path.join(self._cache_dir, file_name)
+        return open(full_path, 'w+')
+
+
+    def get(self, file_name):
+        full_path = os.path.join(self._cache_dir, file_name)
+        if os.path.exists(full_path):
+            return open(full_path, 'rb')
+        return None
+
+
+    def evict(self):
+        data_path = os.path.join(self._cache_dir, 'data')
+        shutil.rmtree(data_path)
+        self._create_cache_structure()
+
+
+
+class Fetcher:
```

Comment:
Hmmm I don't fully agree in principle. A Fetcher fetches files, not stores them. A Cache object stores objects, but is not in charge of anything else. What I think you are proposing is multipurpose Fetcher-Cache-class that does both functions.
However they have clearly different functions (fetch files and store files), so why merging them in a single class?
On a sightly different matter, if this library is for people (or ourselves) to debug, isn't it better to provide independent classes that deal only with a concrete functionality? That way you can abstractly analyse (for example) your cache with a cache object, without caring about any Fetcher. Because... who would imagine that the fetcher acts also as a Cache? Not so intuitive, I think